spec/features/create_etd_spec.rb

```diff
@@ -0,0 +1,235 @@
+    click_link('View Stanford University publication license')
+    page.find('input#cbLicenseStanford').check
+    click_link('Close this window')
+    click_link('View Creative Commons licenses')
+    page.select 'CC Attribution license', from: 'selectCCLicenseOptions'
+    click_link('Close this window')
+
+    # set embargo
+    click_link('Postpone release')
+    page.select '6 months', from: 'selectReleaseDelayOptions'
+    click_link('Close this window')
+
+    expect(page.find('#pbRightsSelected')['style']).to match(/background-image/)
+
+    # "submit etd to registrar"
+    accept_alert do
+      page.find('#submitToRegistrar').click # javascript
+    end
+    # page.find waits for this element to appear
+    expect(page.find('#submissionSuccessful')).to have_content('Submission successful')
+    expect(page.find('#submitToRegistrarDiv > p.progressItemChecked')).to have_content('Submitted')
+
+    # fake reader approval
+    reader_progress_list_el = all('#progressBoxContent > ol > li')[8]
+    expect(reader_progress_list_el).to have_content('Verified by Final Reader - Not done')
+    reader_approved = xml_from_registrar.dup.sub(/<readerapproval>Not Submitted<\/readerapproval>/,
+                                                 '<readerapproval>Approved</readerapproval>')
+    now = Time.now.strftime('%m/%d/%Y %T')
+    reader_approved.sub!(/<readeractiondttm> <\/readeractiondttm>/, "<readeractiondttm>#{now}</readeractiondttm>")
+    reader_approved.sub!(/<readercomment> <\/readercomment>/, '<readercomment>Spock approves</readercomment>')
+    resp_body = simulate_registrar_post(reader_approved)
+    expect(resp_body).to eq "#{prefixed_druid} updated"
+    page.refresh # needed to show updated progress box
+    reader_progress_list_el = all('#progressBoxContent > ol > li')[8]
+    expect(reader_progress_list_el).to have_content('Verified by Final Reader - Done')
+
+    # fake registrar approval
+    registrar_progress_list_el = all('#progressBoxContent > ol > li')[9]
+    expect(registrar_progress_list_el).to have_content('Approved by Registrar - Not done')
+    registrar_approved = xml_from_registrar.dup.sub(/<regapproval>Not Submitted<\/regapproval>/,
+                                                    '<regapproval>Approved</regapproval>')
+    now = Time.now.strftime('%m/%d/%Y %T')
+    registrar_approved.sub!(/<regactiondttm> <\/regactiondttm>/, "<regactiondttm>#{now}</regactiondttm>")
+    resp_body = simulate_registrar_post(registrar_approved)
+    expect(resp_body).to eq "#{prefixed_druid} updated"
+    page.refresh # needed to show updated progress box
+    registrar_progress_list_el = all('#progressBoxContent > ol > li')[9]
+    expect(registrar_progress_list_el).to have_content('Approved by Registrar - Done')
+
+    expect(page.find('#submissionApproved')).to have_content('Submission approved')
+
+    # check Argo for object
+    sleep(3) # waiting for Fedora/Solr
```

Comment:
We use a pattern in other specs where we retry until the visit is successful. I think that will be more consistent than this.

Reply:
cool.  Then we can wait for the embargoed text to turn up as well.